scheduler/scheduler.go

```diff
@@ -139,22 +146,87 @@ func spawnHosts(ctx context.Context, newHostsNeeded map[string]int) (map[string]
 			continue
 		}
 
+		if ctx.Err() != nil {
+			return nil, errors.New("scheduling run canceled.")
+		}
+
+		d, err := distro.FindOne(distro.ById(distroId))
```

Comment:
we should in general, not do findOnes in loops, better to do in queries

Reply:
there is actually a check before this that makes sure there is only one distro in the map, so this for loop currently doesn't ever run more than the once. should i just take out the loop entirely?

Reply:
also for your question above, "hasContainers" isn't always the same as parentID != "" because it is also empty when it is a regular host that isn't a parent or container

Reply:
take out the loop

Reply:
the loop will be removed in EVG-3350 so I think this is fine currently